src/main/java/meteordevelopment/meteorclient/MeteorClient.java

```diff
@@ -144,38 +145,30 @@ private void openGui() {
     private static void init(InitStage initStage) {
         Reflections reflections = new Reflections("meteordevelopment.meteorclient", Scanners.MethodsAnnotated);
         Set<Method> initTasks = reflections.getMethodsAnnotatedWith(Init.class);
-        if (initTasks == null || initTasks.size() < 1) return;
+        if (initTasks == null) return;
+        Map<Class<?>, List<Method>> byClass = initTasks.stream()
+            .collect(Collectors.groupingBy(Method::getDeclaringClass));
+        Set<Method> left = new HashSet<>(initTasks);
 
-        for (Method initTask : initTasks) {
-            Init annotation = initTask.getAnnotation(Init.class);
-            if (annotation != null && annotation.stage().equals(initStage)) reflectInit(initTask, annotation);
+        for (Method m; (m = left.stream().findAny().orElse(null)) != null;) {
+            reflectInit(m, initStage, left, byClass);
         }
     }
 
-    private static void reflectInit(Method task, Init annotation) {
-        Class<?>[] preTasks = annotation.dependencies();
-
-        try {
-            if (preTasks == null || preTasks.length < 1) {
-                task.invoke(null);
-            }
-            else {
-                for (Class<?> aClass : preTasks) {
-                    Set<Method> preInits = new Reflections(aClass).getMethodsAnnotatedWith(Init.class);
```

Comment:
This line does not work, its just not how you use Reflections, it will always return null
Also if it did it would cause initialize to be called a second time